File: src/expansion/scala/mrtjp/projectred/expansion/item/BatteryItem.scala
package mrtjp.projectred.expansion.item

import mrtjp.projectred.expansion.ExpansionContent
import net.minecraft.item.Item

class BatteryItem extends Item(new Item.Properties()
        .tab(ExpansionContent.expansionItemGroup)
        .stacksTo(1)
        .durability(1600)
        .setNoRepair()) with TChargableBatteryItem
{
    override def isEmpty = false

    override def getEmptyVariant:Item = ExpansionContent.emptyBatteryItem.get()
    override def getChargedVariant:Item = this
}

File: expansion/src/main/java/mrtjp/projectred/expansion/item/BatteryItem.java
package mrtjp.projectred.expansion.item;

import mrtjp.projectred.expansion.init.ExpansionItems;
import net.minecraft.world.item.Item;

public class BatteryItem extends Item implements IRechargableBattery {

    public BatteryItem() {
        super(new Item.Properties()
                .durability(1600)
                .setNoRepair());
    }

    @Override
    public Item getChargedVariant() {
        return this;
    }

    @Override
    public Item getEmptyVariant() {
        return ExpansionItems.EMPTY_BATTERY_ITEM.get();
    }
}
